### DISCUSSION TITLE
[CSV Importer] Import data structure from Firefly export

### DISCUSSION THREAD

**Discussion by AlexKalopsia** [firefly-iii/firefly-iii#4659] [Firefly III]
I run an export directly from Firefly, and I am now having some issues importing it.  I assumed CSV Importer was ready to quickly import from the files, but that is seemingly not the case.

Let's begin with the date format:  

The export dates look like `2021-04-13T11:21:26+02:00`. My solution was to use `Y-m-dTH:i:sP`, but this required a bit of research. Would be good to provide this info somewhere on the page or in the docs.

CSV Export structure:

* `user_id`: I don't think I can pair it to any of the available options
* `group_id`: same. I am not sure what it corresponds to
* `journal_id`: same as above
* `created_date`: don't think I can import this info? 
* `updated_date`: same as above
* `group_title`: not sure what this is, but feels like the column should be next to `group_id`?
* `type`: I don't think I can match it with any of the options?
* `amount`: matched with `Amount`
* `foreign_amount`: matched with `Amount (in foreign currency)`
* `currency_code`: matched with `Currency code (ISO 4217)`
* `foreign_currency_code`: matched with Foreign currency code (ISO 4217)`
* `description`: matched with `Description`
* `date`: matched with `Date`, though I am unsure how it differs from `Transition invoice date`
* `source_name`: matched with `Asset account (name)`. I am very unsure about this, since some sources are not asset accounts.
* `source_iban`: matched with `Asset account (IBAN)`. Again, is Asset the correct choice here?
* `source_type`: I don't think this can be matched with anything? Sounds like a pretty big issue, connected to the two points above
* `destination_name`: matched with `Opposing account (name)`
* `destination_iban`: matched with `Opposing account (IBAN)`
* `destination_type`: cannot match this. Same problem as with source account.
* `reconciled`: cannot match this with anything
* `category`: matched with `Category name`
* `budget`: matched with `Budget name`
* `bill`: matched with `Bill name`
* `tags`: matched with `Tags (comma separated)`
* `notes`: matched with `Note(s)`
* `sepa_XX`: matched with all the `SEPA` options
* `external_uri`: not sure here. Is `External ID` correct?

the remaining columns seem somewhat unimportant

In general it feels a bit of a shame that I can't very quickly do an export-import from Firefly into Firefly (ie from one instance to another). 

Would be super helpful to understand a bit better how to match those columns that I was unsure about. 

Thanks

**Comment by AlexKalopsia** [comment]
Just to clarify, I have found the official config here https://github.com/firefly-iii/import-configurations/blob/main/firefly-iii/5.3.0.json
but I still have core issues relative specifically to mapping Asset accounts / Opposing Accounts: it seems like there is - in fact - no option to map `source_id` to any of the Revenue Accounts, so I am completely unable to proceed with the import.

**Comment by JC5** [comment]
Why would you want to export Firefly III data and then immediately import it?

**Reply by AlexKalopsia** [reply]
Well, "immediately" is not a requirement here. In my specific case I was setting up a separate FF instance, holding only a portion of the data.

So to me it feels like anything that FF natively exports, should be seamlessly imported by FF as well.

**Reply by JC5** [reply]
No, sorry. That's what the database itself is way better equipped for.

**Reply by AlexKalopsia** [reply]
Sure, working directly on the db will allow more precise export, but in terms of user experience that is quite a lot more work for the user (being a Firefly user doesn't necessarily imply familiarity with SQL queries).

But again, here I am not arguing that it needs to be a fully fledged import/export tool, but just something that feels reliable and intuitive.

In other words, I don't think I have ever seen a tool that can export data but can't smoothly import it (especially when it was the tool itself doing the original export).

The question regarding Revenue/Opposing accounts still stands, as it makes it impossible to import the data FF exported.

**Reply by JC5** [reply]
Sorry, but in terms of software development, it's an insane amount of work that can be replicated by a database in 5 minutes tops. 

So the export is meant as an export. To be imported into other systems or tools. 

I can't vouch for other peoples skills and what people can or can't do isn't my problem. 

Match the account names like you suggested and try. You can map accounts if you want to.

Otherwise, I fear that the export routine isn't going to help for your particular use case. I've added a little disclaimer to the documentation.

**Reply by AlexKalopsia** [reply]
Again, I am not saying it's in the scope of this to be a full-on advanced import/export tool, but what you are saying is that the export tool can create exports that can be imported into other systems or tools [that are not Firefly]? To me that is still a very big shortcoming, but I understand we are not going to agree on this. 

What I find confusing or still unclear is how come FF is capable to translate its data into a csv file, but it is not capable to read that very same data.

Can I not map a `source_name` with a Revenue account? From the drop down I can only select an Asset account (which is not what I need) or an Opposing account (which only shows me the list of Expense accounts)
